python/wasmedge_sock/wasi_socket_ext.c

```diff
@@ -460,6 +461,61 @@ int bind(int fd, const struct sockaddr* addr, socklen_t len) {
   return res;
 }
 
+int socketpair(int domain, int type, int protocol, int sv[2]) {
+  WSEDEBUG("WSME| socketpair() domain=%d type=%d _protocol=%d\n", domain, type,
+         _protocol);
+
+  int32_t err;
+  int lsock = socket(domain, type, protocol);
+
+  we_address_buffer raw;
+  raw.family = utw_sock_domain(domain);
+  we_address addr = {.buf = &raw, .len = 128};
+  uint16_t port = 0;
+
+  err = __import_wasip1_sock_bind(lsock, &addr, port);
+  if (err != 0) {
+    errno = err;
+    return -1;
+  }
+
+  int rl = listen(lsock, 1);
+  if (rl != 0) {
+      errno = rl;
+      return -1;
+  }
+
+  uint32_t lport;
+  we_address_buffer lraw;
+  we_address laddr = {.buf = &lraw, .len = sizeof(lraw)};
+
+  err = __import_wasip1_sock_getlocaladdr(lsock, &laddr, &lport);
+  if (err != 0) {
+    errno = err;
+    return -1;
+  }
+
+  int csock = socket(domain, type, protocol);
+
+  err = __import_wasip1_sock_connect(csock, &laddr, lport);
```

Comment:
Good question. I went that way because I was hitting issues with the higher level APIs but these issues were most certainly from the unnamed sockets issue. I can look at using those back.